app/api/route/models/base_api_model.py

```diff
@@ -0,0 +1,12 @@
+from typing import ClassVar, Type
+
+from marshmallow import Schema  # noqa: F401
+from marshmallow_dataclass import dataclass as marshmallow_dataclass
+
+
+@marshmallow_dataclass
+class BaseApiModel:
+    # Adding this to the base model helps
+    # mypy recognize the .Schema that is dynamically
+    # added by marshmallow_dataclass
+    Schema: ClassVar[Type[Schema]] = Schema  # noqa: F811
```

Comment:
i took a look at [how marshmallow dataclasses implements adding the `Schema` attribute to the class object](https://github.com/lovasoa/marshmallow_dataclass/blob/master/marshmallow_dataclass/__init__.py#L212-L219), it looks like in their type hint they are not actually recognizing that the return value from the decorator is a different type with an extra attribute:
```
def decorator(clazz: Type[_U]) -> Type[_U]:
```
IMO i think this is bad design by the marshmallow dataclass library. instead of using `Model.Schema`. let's instead have the schema be separate, like this:

```
# user
import marshmallow_dataclass

class User:
  ...

UserSchema = marshmallow_dataclass.class_schema(User)
```

then in places that need the schema, they'll import `UserSchema`

Reply:
Yeah, that seems fair. Might help to distinguish that there are in fact TWO classes for every class here, the auto-generated Schema, and the dataclass you work with.

Reply:
Similar to the issue described elsewhere, I'm not sure I can actually do this and maintain inheritance. If I define a `@dataclass` for the model then it runs into several issues with fields being defined out of order regarding the defaults. There might be a way to work around this, but I'm not sure I have the bandwidth to try and restructure that at this time.